designs/native-ami-versioning.md

```diff
@@ -0,0 +1,121 @@
+
+```
+spec:
+    amiSelector:
+        aws-ids: "ami-123,ami-456"
+```
+
+Once the user wants to return to the latest release, they’ll have to restore their configurations to the previous default as the following.
+
+```
+spec:
+    amiFamily: Ubuntu # Or Bottlerocket or AL2
+```
+
+While this isn’t a difficult process, it isn't viable to require users to jump through these hoops. They must be aware of new EKS Optimized AMI releases, understand SSM aliases to fetch the AMI IDs, and modify their AWSNodeTemplate values in their CI/CD pipelines multiple times. Some users have [requested these improvements](https://github.com/aws/karpenter/issues/1495). In order to improve the experience for all users, we’ll introduce (1) Bake Time and (2) Release Version Pinning improvements.
+
+## Bake Time
+
+### 🔑 Introduce a minimumAgeDays field in the amiSelector.
+
+While this hasn’t been asked for explicitly, bake time is a common concept in phasing in new changes. Whether a user has a mandated bake time for production or developers have an estimated time to validate workloads in before rolling out to prod, bake time provides a sliding window of leeway. Users can be cautious and set a high value, or leave out the setting for no bake time requirements.
+
+```
+apiVersion: karpenter.k8s.aws/v1alpha1
+kind: AWSNodeTemplate
+metadata:
+  name: default
+spec:
+  amiSelector:
+    minimumAgeDuration: 10
+```
+
+### How `minimumAgeDuration` interacts with the other methods of AMI selection
+
+#### *EKS Optimized AMI (no other AMI Selector field)*
+
+1. The newest EKS Optimized AMI that fits the `minimumAgeDuration` selector will be chosen.
+2. An AMI will be considered eligible at the time and day that it was created as shown in aws ec2 describe-images.
+
+#### *Arbitrary tags*
+
+1. An AMI tagged and discovered via the amiSelector tags will not be chosen unless it fits the `minimumAgeDuration` filter.
+2. This has the downside of resulting in no AMIs found if users are too restrictive with tags and the `minimumAgeDuration` filter. The upside is that users will know any AMI created before x days ago will not be included.
+
+#### *AMI IDs*
+
+1. When using AMI IDs, other tags specified are not respected. To keep explicit AMI ID usage as an override, using `minimumAgeDuration` and AMI IDs will result in `minimumAgeDuration` not being respected.
+
+### DriftTTL in the ProvisionerSpec
+
+Karpenter could include a drift TTL in the Provisioner. Nodes would have to be drifted for the TTL's duration to be deprovisioned. This can implement bake time, but will also apply for all future methods of Drift, such as zonal and instance permission requirements. Since other Drift methods may need more immediate actions when it occurs, setting a Drift TTL may not be granular as needed. This
```

Comment:
```suggestion
Karpenter could include a drift TTL in the Provisioner. Nodes would have to be drifted for the TTL's duration to be deprovisioned. This can implement bake time, but will also apply for all future methods of Drift, such as zonal and instance permission requirements. Since other Drift methods may need more immediate actions when it occurs, setting a Drift TTL may not be granular as needed.
```